org.eclipse.winery.edmm/src/main/java/org/eclipse/winery/edmm/model/EdmmImporter.java

```diff
@@ -0,0 +1,458 @@
+        return importEdmmModel(DeploymentModel.of(edmmYaml), override);
+    }
+
+    private boolean importEdmmModel(DeploymentModel deploymentModel, boolean override) {
+        logger.info("Starting to import \"{}\"", deploymentModel.getName());
+
+        EntityGraph deploymentModelGraph = deploymentModel.getGraph();
+        Optional<Entity> componentTypesOptional = deploymentModelGraph.getEntity(EntityGraph.COMPONENT_TYPES);
+        if (componentTypesOptional.isPresent()) {
+            Set<Entity> componentTypes = componentTypesOptional.get().getChildren();
+            logger.debug("Found {} Component Types to import", componentTypes.size());
+
+            componentTypes.forEach(this::importComponentTypes);
+        }
+
+        Optional<Entity> relationTypesOptional = deploymentModelGraph.getEntity(EntityGraph.RELATION_TYPES);
+        if (relationTypesOptional.isPresent()) {
+            Set<Entity> relationTypes = relationTypesOptional.get().getChildren();
+            logger.debug("Found {} Relation Types to import", relationTypes.size());
+
+            relationTypes.forEach(this::importRelationTypes);
+        }
+
+        Set<RootComponent> components = deploymentModel.getComponents();
+        logger.debug("Found {} Components to import", components.size());
+        importEdmmApplication(deploymentModel, components, override);
+
+        return true;
+    }
+
+    private void importRelationTypes(Entity entity) {
+        String typeName = entity.getName();
+
+        QName qName = existsQNameForType(typeName, relationshipTypesString);
+
+        if (qName == null) {
+            logger.debug("Creating new Relationship Type \"{}\"", typeName);
+            qName = EdmmUtils.getQNameFromType(typeName, relationshipTypesString);
+            TRelationshipType.Builder builder = new TRelationshipType.Builder(qName);
+
+            importTypeSpecificElements(entity, builder, relationshipTypesString);
+
+            RelationshipTypeId relationshipTypeId = new RelationshipTypeId(qName);
+            try {
+                repository.setElement(relationshipTypeId, builder.build());
+            } catch (IOException e) {
+                logger.error("Could not create Relationship Type with QName: \"{}\"", relationshipTypeId.getQName());
+            }
+
+            logger.debug("Created new Relationship Type \"{}\"", qName);
+        }
+    }
+
+    private void importEdmmApplication(DeploymentModel deploymentModel, Set<RootComponent> components, boolean override) {
+        String deploymentModelName = deploymentModel.getName() != null
+            ? deploymentModel.getName().replaceAll(".y(a?)ml", "")
+            : "Imported-EDMM_" + System.currentTimeMillis();
+
+        ServiceTemplateId serviceTemplateId = new ServiceTemplateId(
+            new QName(EdmmUtils.IMPORTED_EDMM_NAMESPACE + "serviceTemplates", deploymentModelName)
+        );
+        if (repository.exists(serviceTemplateId) && !override) {
```

Comment:
so the overwrite flag only applies to the whole servicetemplate. If the flag is false, and we only have, e.g., a node type, that has the same name but with different properties, then we still overwrite it. Correct?

Reply:
So if the importer already finds a corresponding type, it is currently not overwritten.